### ISSUE TITLE
TLS error while using both AWS-IoT shadow and topic services (IDFGH-4132)

### ISSUE THREAD

**Issue by rhr407** [espressif/esp-idf#5999]
----------------------------- Delete below -----------------------------

**Reminder: If your issue is a general question, starts similar to "How do I..", or is related to 3rd party development kits/libs, please discuss this on our community forum at https://esp32.com instead.**

INSTRUCTIONS
============

Before submitting a new issue, please follow the checklist and try to find the answer.

- [x] I have read the documentation [ESP-IDF Programming Guide](https://docs.espressif.com/projects/esp-idf/en/latest/) and the issue is not addressed there.
- [x] I have updated my IDF branch (master or release) to the latest version and checked that the issue is present there.
- [x] I have searched the issue tracker for a similar issue and not found a similar issue.

If the issue cannot be solved after the steps before, please follow these instructions so we can get the needed information to help you in a quick and effective fashion.

1. Fill in all the fields under **Environment** marked with [ ] by picking the correct option for you in each case and deleting the others.
2. Describe your problem.
3. Include [debug logs from the "monitor" tool](https://docs.espressif.com/projects/esp-idf/en/latest/get-started/idf-monitor.html#automatically-decoding-addresses), or [coredumps](https://docs.espressif.com/projects/esp-idf/en/latest/api-guides/core_dump.html).
4. Providing as much information as possible under **Other items if possible** will help us locate and fix the problem.
5. Use [Markdown](https://guides.github.com/features/mastering-markdown/) (see formatting buttons above) and the Preview tab to check what the issue will look like.
6. Delete these instructions from the above to the below marker lines before submitting this issue.

**IMPORTANT: If you do not follow these instructions and provide the necessary details, your issue may not be resolved.**

----------------------------- Delete above -----------------------------

## Environment

- Development Kit:      [ESP32-DevKitC]
- Kit version (for DevKitC): [v3]
- Module or chip used:  [ESP32-WROOM-32|]
- IDF version (run ``git describe --tags`` to find it):
    // v4.3-dev-1197-g8bc19ba89
- Build System:         [Make|CMake|idf.py]
- Compiler version (run ``xtensa-esp32-elf-gcc --version`` to find it):
    // 1.22.0-80-g6c4433a
- Operating System:     [Linux]
- Using an IDE?: [No]
- Power Supply:         [USB]

## Problem Description

I am using AWS-IoT component with shadow and topic service simultaneously. It works fine for some time but one error arrives ad given below:
```
I (964511) wifi:bcn_timout,ap_probe_send_start
I (966681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (967011) wifi:ap_probe_send over, resett wifi status to disassoc
I (967011) wifi:state: run -> init (c800)
I (967011) wifi:pm stop, total sleep time: 26345448 us / 41062945 us

I (967021) wifi:new:<4,0>, old:<4,0>, ap:<255,255>, sta:<4,0>, prof:1
I (967021) BLE_PROV: Disconnected. Connecting to the AP again...
E (967031) aws_iot: failed! mbedtls_ssl_handshake returned -0x2700
E (967041) aws_iot:     Unable to verify the server's certificate. 
I (967151) wifi:new:<4,0>, old:<4,0>, ap:<255,255>, sta:<4,0>, prof:1
I (967151) wifi:state: init -> auth (b0)
I (967161) wifi:state: auth -> assoc (0)
I (967171) wifi:state: assoc -> run (10)
I (967201) wifi:connected with TALKTALK4B89FC, aid = 1, channel 4, BW20, bssid = d0:c6:5b:4b:8a:0c
I (967201) wifi:security: WPA2-PSK, phy: bgn, rssi: -61
I (967211) wifi:pm start, type: 1

I (967211) wifi:AP's beacon interval = 102400 us, DTIM period = 1
```
### Expected Behavior
No such error
### Code to reproduce this issue

```cpp
#include "esp_log.h"

#include "aws.h"

#include <stdlib.h>

char HOST[255] = AWS_IOT_MQTT_HOST;

AWS_IoT_Client mqttClient;

char uart_message_Arrived = false;

/**
 * @brief Default MQTT PORT is pulled from the aws_iot_config.h
 */
uint32_t PORT = AWS_IOT_MQTT_PORT;

static const char *TAG = "aws-shadow";

static char *T;


static char temp_changed = true;

static bool shadowUpdateInProgress;
uint8_t* DATA;


static void get_device_MAC(char *mac, size_t max)
{
    uint8_t eth_mac[6];
    esp_wifi_get_mac(WIFI_IF_STA, eth_mac);
    snprintf(mac, max, "%02X%02X%02X%02X%02X%02X", eth_mac[0], eth_mac[1], eth_mac[2], eth_mac[3], eth_mac[4], eth_mac[5]);
    printf("MAC: %s\n", mac);
}


void shadow_callback(const char *pThingName, ShadowActions_t action, Shadow_Ack_Status_t status,
                     const char *pReceivedJsonDocument, void *pContextData) {

    IOT_UNUSED(pThingName);
    IOT_UNUSED(action);
    IOT_UNUSED(pReceivedJsonDocument);
    IOT_UNUSED(pContextData);

    shadowUpdateInProgress = false;

    if (SHADOW_ACK_TIMEOUT == status) {
        ESP_LOGE(TAG, "Update timed out");
    } else if (SHADOW_ACK_REJECTED == status) {
        ESP_LOGE(TAG, "Update rejected");
    } else if (SHADOW_ACK_ACCEPTED == status) {
        ESP_LOGI(TAG, "Update accepted");
    }
}



void iot_subscribe_callback_handler(AWS_IoT_Client *pClient, char *topicName, uint16_t topicNameLen,
                                    IoT_Publish_Message_Params *params, void *pData) {
    ESP_LOGI(TAG, "Subscribe callback");
    ESP_LOGI(TAG, "%.*s\t%.*s", topicNameLen, topicName, (int) params->payloadLen, (char *)params->payload);
}

void disconnectCallbackHandler(AWS_IoT_Client *pClient, void *data) {
    ESP_LOGW(TAG, "MQTT Disconnect");
    IoT_Error_t rc = FAILURE;

    if (NULL == pClient) {
        return;
    }

    if (aws_iot_is_autoreconnect_enabled(pClient)) {
        ESP_LOGI(TAG, "Auto Reconnect is enabled, Reconnecting attempt will start now");
    } else {
        ESP_LOGW(TAG, "Auto Reconnect not enabled. Starting manual reconnect...");
        rc = aws_iot_mqtt_attempt_reconnect(pClient);
        if (NETWORK_RECONNECTED == rc) {
            ESP_LOGW(TAG, "Manual Reconnect Successful");
        } else {
            ESP_LOGW(TAG, "Manual Reconnect Failed - %d", rc);
        }
    }
}



void setTemp_Callback(const char *pJsonString, uint32_t JsonStringDataLen, jsonStruct_t *pContext) {
    IOT_UNUSED(pJsonString);
    IOT_UNUSED(JsonStringDataLen);

    if (pContext != NULL) {
        ESP_LOGI(TAG, "Delta - Temperature state changed to %f", *(float *) (pContext->pData));
        temp_changed = true;
    }

}

void aws_shadow(void *param) {
    IoT_Error_t rc = FAILURE;

    static char TOPIC[100];
    static int TOPIC_LEN;
    char cPayload[200];

    // unsigned int DT, S;

    // float P, U , LO, LA, F, E, A;


    // DT = 0;
    // T = 0;
    // S = 0;

    // P = 0;
    // U = 0;
    // LO = 0;
    // LA = 0;

    // F = 0;
    // E = 0;
    // A = 0;


    IoT_Client_Init_Params mqttInitParams = iotClientInitParamsDefault;
    IoT_Client_Connect_Params connectParams = iotClientConnectParamsDefault;

    IoT_Publish_Message_Params paramsQOS1;


    mqttInitParams.enableAutoReconnect = true; // We enable this later below
    mqttInitParams.pHostURL = AWS_IOT_MQTT_HOST;
    mqttInitParams.port = AWS_IOT_MQTT_PORT;

#if defined(CONFIG_EMBEDDED_CERTS)
    mqttInitParams.pRootCALocation = (const char *)aws_root_ca_pem_start;
    mqttInitParams.pDeviceCertLocation = (const char *)certificate_pem_crt_start;
    mqttInitParams.pDevicePrivateKeyLocation = (const char *)private_pem_key_start;

#elif defined(CONFIG_FILESYSTEM_CERTS)
    mqttInitParams.pRootCALocation = ROOT_CA_PATH;
    mqttInitParams.pDeviceCertLocation = DEVICE_CERTIFICATE_PATH;
    mqttInitParams.pDevicePrivateKeyLocation = DEVICE_PRIVATE_KEY_PATH;
#endif

    mqttInitParams.mqttCommandTimeout_ms = 20000;
    mqttInitParams.tlsHandshakeTimeout_ms = 5000;
    mqttInitParams.isSSLHostnameVerify = true;
    // mqttInitParams.disconnectHandler = disconnectCallbackHandler;
    mqttInitParams.disconnectHandler = NULL;
    mqttInitParams.disconnectHandlerData = NULL;

#ifdef CONFIG_SDCARD_CERTS
    ESP_LOGI(TAG, "Mounting SD card...");
    sdmmc_host_t host = SDMMC_HOST_DEFAULT();
    sdmmc_slot_config_t slot_config = SDMMC_SLOT_CONFIG_DEFAULT();
    esp_vfs_fat_sdmmc_mount_config_t mount_config = {
        .format_if_mount_failed = false,
        .max_files = 3,
    };
    sdmmc_card_t* card;
    esp_err_t ret = esp_vfs_fat_sdmmc_mount("/sdcard", &host, &slot_config, &mount_config, &card);
    if (ret != ESP_OK) {
        ESP_LOGE(TAG, "Failed to mount SD card VFAT filesystem. Error: %s", esp_err_to_name(ret));
        abort();
    }
#endif

    // rc = aws_iot_mqtt_init(&mqttClient, &mqttInitParams);
    // if (SUCCESS != rc) {
    //     ESP_LOGE(TAG, "aws_iot_mqtt_init returned error : %d ", rc);
    //     abort();
    // }


    connectParams.keepAliveIntervalInSec = 600;
    connectParams.isCleanSession = true;
    connectParams.MQTTVersion = MQTT_3_1_1;
    /* Client ID is set in the menuconfig of the example */
    connectParams.pClientID = CONFIG_AWS_CLIENT_ID;
    connectParams.clientIDLen = (uint16_t) strlen(CONFIG_AWS_CLIENT_ID);
    connectParams.isWillMsgPresent = false;




    char JsonDocumentBuffer[MAX_LENGTH_OF_UPDATE_JSON_BUFFER];
    size_t sizeOfJsonDocumentBuffer = sizeof(JsonDocumentBuffer) / sizeof(JsonDocumentBuffer[0]);

    float setTemp = 0.0;
    float currentTemp = 0.0;
    float prev_currentTemp = 0.0;

    static char sendTempUART[12];



    char snum[5];


    char mac[20] ;
    get_device_MAC(mac, sizeof(mac));

    sprintf(TOPIC, "%s%s%s", "dt/coreone/cognitoUserID/", mac, "/all");


    // sprintf(TOPIC, "%s", "dt/coreone/cognitoUserID/deviceID/all");

    printf("TOPIC is %s\n", TOPIC);

    TOPIC_LEN = strlen(TOPIC);

    printf("TOPIC_LEN : %d\n", TOPIC_LEN);


    connectParams.will.pTopicName = TOPIC;
    connectParams.will.topicNameLen = strlen(TOPIC);

    connectParams.will.pMessage = "{\"state\":{\"reported\":{\"online\":false}}}";
    connectParams.will.msgLen = strlen("{\"state\":{\"reported\":{\"online\":false}}}");
    connectParams.will.qos = QOS1; //QoS 2 is not supported by aws-IoT SDK, we put '1' here.



    paramsQOS1.qos = QOS1;
    paramsQOS1.payload = (void *) cPayload;
    paramsQOS1.isRetained = 0;


    jsonStruct_t setTempHandler;
    setTempHandler.cb = setTemp_Callback;
    setTempHandler.pKey = "setTemp";
    setTempHandler.pData = &setTemp;
    setTempHandler.type = SHADOW_JSON_FLOAT;
    setTempHandler.dataLength = sizeof(float);

    jsonStruct_t currentTempHandler;
    currentTempHandler.cb = NULL;
    currentTempHandler.pKey = "currentTemp";
    currentTempHandler.pData = &currentTemp;
    currentTempHandler.type = SHADOW_JSON_FLOAT;
    currentTempHandler.dataLength = sizeof(float);

    // initialize the mqtt client
    // AWS_IoT_Client mqttClient;

    ShadowInitParameters_t sp = ShadowInitParametersDefault;
    sp.pHost = AWS_IOT_MQTT_HOST;
    sp.port = AWS_IOT_MQTT_PORT;

#if defined(CONFIG_EMBEDDED_CERTS)
    sp.pClientCRT = (const char *)certificate_pem_crt_start;
    sp.pClientKey = (const char *)private_pem_key_start;
    sp.pRootCA = (const char *)aws_root_ca_pem_start;
#elif defined(CONFIG_FILESYSTEM_CERTS)
    sp.pClientCRT = DEVICE_CERTIFICATE_PATH;
    sp.pClientKey = DEVICE_PRIVATE_KEY_PATH;
    sp.pRootCA = ROOT_CA_PATH;
#endif
    sp.enableAutoReconnect = false;
    sp.disconnectHandler = NULL;

#ifdef CONFIG_SDCARD_CERTS
    ESP_LOGI(TAG, "Mounting SD card...");
    sdmmc_host_t host = SDMMC_HOST_DEFAULT();
    sdmmc_slot_config_t slot_config = SDMMC_SLOT_CONFIG_DEFAULT();
    esp_vfs_fat_sdmmc_mount_config_t mount_config = {
        .format_if_mount_failed = false,
        .max_files = 3,
    };
    sdmmc_card_t* card;
    esp_err_t ret = esp_vfs_fat_sdmmc_mount("/sdcard", &host, &slot_config, &mount_config, &card);
    if (ret != ESP_OK) {
        ESP_LOGE(TAG, "Failed to mount SD card VFAT filesystem. Error: %s", esp_err_to_name(ret));
        abort();
    }
#endif


    ESP_LOGI(TAG, "Shadow Init");
    rc = aws_iot_shadow_init(&mqttClient, &sp);
    if (SUCCESS != rc) {
        ESP_LOGE(TAG, "aws_iot_shadow_init returned error %d, aborting...", rc);
        abort();
    }

    ShadowConnectParameters_t scp = ShadowConnectParametersDefault;
    scp.pMyThingName = CONFIG_AWS_THING_NAME;
    scp.pMqttClientId = CONFIG_AWS_CLIENT_ID;
    scp.mqttClientIdLen = (uint16_t) strlen(CONFIG_AWS_CLIENT_ID);

    ESP_LOGI(TAG, "Shadow Connect");
    rc = aws_iot_shadow_connect(&mqttClient, &scp);
    if (SUCCESS != rc) {
        ESP_LOGE(TAG, "aws_iot_shadow_connect returned error %d, aborting...", rc);
        abort();
    }

    /*
     * Enable Auto Reconnect functionality. Minimum and Maximum time of Exponential backoff are set in aws_iot_config.h
     *  #AWS_IOT_MQTT_MIN_RECONNECT_WAIT_INTERVAL
     *  #AWS_IOT_MQTT_MAX_RECONNECT_WAIT_INTERVAL
     */
    rc = aws_iot_shadow_set_autoreconnect_status(&mqttClient, true);
    if (SUCCESS != rc) {
        ESP_LOGE(TAG, "Unable to set Auto Reconnect to true - %d, aborting...", rc);
        abort();
    }

    rc = aws_iot_shadow_register_delta(&mqttClient, &setTempHandler);

    if (SUCCESS != rc) {
        ESP_LOGE(TAG, "Shadow Register Delta Error");
    }




// =========================================================================


    // loop and publish a change in temperature
    while (NETWORK_ATTEMPTING_RECONNECT == rc || NETWORK_RECONNECTED == rc || SUCCESS == rc) {

        rc = aws_iot_shadow_yield(&mqttClient, 200) ;

        if (NETWORK_ATTEMPTING_RECONNECT == rc || shadowUpdateInProgress) {
            rc = aws_iot_shadow_yield(&mqttClient, 1000) ;

            // If the client is attempting to reconnect, or already waiting on a shadow update,
            // we will skip the rest of the loop.
            continue;
        }

        if (uart_message_Arrived) {

            uart_message_Arrived = false;

            sprintf(cPayload, (char*)DATA);


            paramsQOS1.payload = cPayload;
            paramsQOS1.payloadLen = strlen(cPayload);
            rc = aws_iot_mqtt_publish(&mqttClient, TOPIC, TOPIC_LEN, &paramsQOS1);


            if (rc != SUCCESS) {
                ESP_LOGW(TAG, "No Message Published with rc = %d", rc );
                rc = SUCCESS;
            }
            else   printf("Published\n");

            cJSON * root = cJSON_Parse(cPayload);

            cJSON *temp =  cJSON_GetObjectItem(root, "T");
            T = cJSON_Print(temp);


            currentTemp = atof(T);

            if (prev_currentTemp != currentTemp) {

                temp_changed = 1;
                prev_currentTemp = currentTemp;

            }

        }

        if (temp_changed) {
            rc = aws_iot_shadow_init_json_document(JsonDocumentBuffer, sizeOfJsonDocumentBuffer);
            if (SUCCESS == rc) {
                rc = aws_iot_shadow_add_reported(JsonDocumentBuffer, sizeOfJsonDocumentBuffer, 2, &setTempHandler, &currentTempHandler);
                if (SUCCESS == rc) {
                    rc = aws_iot_finalize_json_document(JsonDocumentBuffer, sizeOfJsonDocumentBuffer);
                    if (SUCCESS == rc) {
                        ESP_LOGI(TAG, "Update Shadow: %s", JsonDocumentBuffer);
                        rc = aws_iot_shadow_update(&mqttClient, CONFIG_AWS_THING_NAME, JsonDocumentBuffer,
                                                   shadow_callback, NULL, 4, true);
                        shadowUpdateInProgress = true;

                    }
                }




                // convert 123 to string [buf]
                itoa(setTemp, snum, 10);


                // gcvt(setTemp, 6, snum);

                sprintf(sendTempUART, "%s%s%s", "{\"T\":", snum, "}");
                sendData(TAG, sendTempUART);



            }

            temp_changed = false;

        }

        vTaskDelay(20 / portTICK_RATE_MS);
    }

// xTaskCreate(&aws_pub, "aws_pub", 8 * 1024, NULL, 6, NULL);

    if (SUCCESS != rc) {
        ESP_LOGE(TAG, "An error occurred in the loop %d", rc);
    }

    ESP_LOGI(TAG, "Disconnecting");
    rc = aws_iot_shadow_disconnect(&mqttClient);

    if (SUCCESS != rc) {
        ESP_LOGE(TAG, "Disconnect error %d", rc);
    }


    ESP_LOGI(TAG, "Shadow Connect");

    rc = aws_iot_shadow_set_autoreconnect_status(&mqttClient, true);
    if (SUCCESS != rc) {
        ESP_LOGE(TAG, "Unable to set Auto Reconnect to true - %d, aborting...", rc);
        abort();
    }

}





```

## Debug Logs

```
Debug log goes here, should contain the backtrace, as well as the reset source if it is a crash.
Please copy the plain text here for us to search the error log. Or attach the complete logs but leave the main part here if the log is *too* long.
```

## Other items if possible

- [ ] sdkconfig file (attach the sdkconfig file from your project folder)
- [ ] elf file in the ``build`` folder (**note this may contain all the code details and symbols of your project.**)
- [ ] coredump (This provides stacks of tasks.)

**Comment by chegewara** [comment]
You are not connected to wifi AP yet:
```
I (967021) BLE_PROV: Disconnected. Connecting to the AP again...
E (967031) aws_iot: failed! mbedtls_ssl_handshake returned -0x2700
E (967041) aws_iot:     Unable to verify the server's certificate. 
I (967151) wifi:new:<4,0>, old:<4,0>, ap:<255,255>, sta:<4,0>, prof:1
I (967151) wifi:state: init -> auth (b0)
I (967161) wifi:state: auth -> assoc (0)
I (967171) wifi:state: assoc -> run (10)
I (967201) wifi:connected with TALKTALK4B89FC, aid = 1, channel 4, BW20, bssid = d0:c6:5b:4b:8a:0c
```

**Comment by rhr407** [comment]
```cpp
Published
I (828881) aws-shadow: Update Shadow: {"state":{"reported":{"setTemp":43.000000,"currentTemp":40.000000}}, "clientToken":"net0001-274"}
I (828911) aws-shadow: Wrote 8 bytes

I (828981) aws-shadow: Update accepted
I (831681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

Published
I (831831) aws-shadow: Update Shadow: {"state":{"reported":{"setTemp":43.000000,"currentTemp":25.000000}}, "clientToken":"net0001-275"}
I (831861) aws-shadow: Wrote 8 bytes

I (831921) aws-shadow: Update accepted
I (834681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

Published
I (834781) aws-shadow: Update Shadow: {"state":{"reported":{"setTemp":43.000000,"currentTemp":40.000000}}, "clientToken":"net0001-276"}
I (834811) aws-shadow: Wrote 8 bytes

I (834871) aws-shadow: Update accepted
I (837681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

Published
I (837951) aws-shadow: Update Shadow: {"state":{"reported":{"setTemp":43.000000,"currentTemp":25.000000}}, "clientToken":"net0001-277"}
I (837981) aws-shadow: Wrote 8 bytes

I (838041) aws-shadow: Update accepted
I (840681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

Published
I (840901) aws-shadow: Update Shadow: {"state":{"reported":{"setTemp":43.000000,"currentTemp":40.000000}}, "clientToken":"net0001-278"}
I (840931) aws-shadow: Wrote 8 bytes

I (840991) aws-shadow: Update accepted
I (843681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

Published
I (843851) aws-shadow: Update Shadow: {"state":{"reported":{"setTemp":43.000000,"currentTemp":25.000000}}, "clientToken":"net0001-279"}
I (843881) aws-shadow: Wrote 8 bytes

I (843941) aws-shadow: Update accepted
I (846681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

Published
I (846811) aws-shadow: Update Shadow: {"state":{"reported":{"setTemp":43.000000,"currentTemp":40.000000}}, "clientToken":"net0001-280"}
I (846841) aws-shadow: Wrote 8 bytes

I (846901) aws-shadow: Update accepted
I (849681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

Published
I (849771) aws-shadow: Update Shadow: {"state":{"reported":{"setTemp":43.000000,"currentTemp":25.000000}}, "clientToken":"net0001-281"}
I (849801) aws-shadow: Wrote 8 bytes

I (849861) aws-shadow: Update accepted
I (852681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

Published
I (852941) aws-shadow: Update Shadow: {"state":{"reported":{"setTemp":43.000000,"currentTemp":40.000000}}, "clientToken":"net0001-282"}
I (852971) aws-shadow: Wrote 8 bytes

I (853061) aws-shadow: Update accepted
I (855681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

Published
I (855891) aws-shadow: Update Shadow: {"state":{"reported":{"setTemp":43.000000,"currentTemp":25.000000}}, "clientToken":"net0001-283"}
I (855921) aws-shadow: Wrote 8 bytes

I (858681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

E (860741) aws-shadow: Update timed out
I (861681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (861911) wifi:bcn_timout,ap_probe_send_start
I (864411) wifi:ap_probe_send over, resett wifi status to disassoc
I (864411) wifi:state: run -> init (c800)
I (864411) wifi:pm stop, total sleep time: 701439439 us / 863194504 us

I (864411) wifi:new:<4,0>, old:<4,0>, ap:<255,255>, sta:<4,0>, prof:1
I (864421) BLE_PROV: Disconnected. Connecting to the AP again...
E (864431) aws_iot: failed! mbedtls_ssl_write returned -0x4e
W (864431) aws-shadow: No Message Published with rc = -12
E (864471) aws_iot: failed! mbedtls_ssl_write returned -0x4e
I (864551) wifi:new:<4,0>, old:<4,0>, ap:<255,255>, sta:<4,0>, prof:1
I (864551) wifi:state: init -> auth (b0)
I (864561) wifi:state: auth -> assoc (0)
I (864561) wifi:state: assoc -> run (10)
I (864601) wifi:connected with TALKTALK4B89FC, aid = 1, channel 4, BW20, bssid = d0:c6:5b:4b:8a:0c
I (864601) wifi:security: WPA2-PSK, phy: bgn, rssi: -64
I (864611) wifi:pm start, type: 1

I (864611) wifi:AP's beacon interval = 102400 us, DTIM period = 1
I (864681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

E (865481) aws_iot: failed! mbedtls_net_connect returned -0x52
I (867681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (867801) BLE_PROV: Connected with IP Address:192.168.1.25
I (867801) esp_netif_handlers: sta ip: 192.168.1.25, mask: 255.255.255.0, gw: 192.168.1.1
I (870681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (873681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

E (873861) aws_iot: failed! mbedtls_ssl_handshake returned -0x6800
I (876681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (879681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (882681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

E (882911) aws_iot: failed! mbedtls_ssl_handshake returned -0x6800
I (885681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (888681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (891681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (894681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (897681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (897951) wifi:bcn_timout,ap_probe_send_start
I (900451) wifi:ap_probe_send over, resett wifi status to disassoc
I (900451) wifi:state: run -> init (c800)
I (900451) wifi:pm stop, total sleep time: 13424467 us / 35843579 us

I (900461) wifi:new:<4,0>, old:<4,0>, ap:<255,255>, sta:<4,0>, prof:1
I (900471) BLE_PROV: Disconnected. Connecting to the AP again...
E (900471) aws_iot: failed! mbedtls_ssl_handshake returned -0x2700
E (900481) aws_iot:     Unable to verify the server's certificate. 
I (900591) wifi:new:<4,0>, old:<4,0>, ap:<255,255>, sta:<4,0>, prof:1
I (900591) wifi:state: init -> auth (b0)
I (900601) wifi:state: auth -> assoc (0)
I (900611) wifi:state: assoc -> run (10)
I (900641) wifi:connected with TALKTALK4B89FC, aid = 1, channel 4, BW20, bssid = d0:c6:5b:4b:8a:0c
I (900641) wifi:security: WPA2-PSK, phy: bgn, rssi: -62
I (900651) wifi:pm start, type: 1

I (900651) wifi:AP's beacon interval = 102400 us, DTIM period = 1
I (900681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (903681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (903801) BLE_PROV: Connected with IP Address:192.168.1.25
I (903801) esp_netif_handlers: sta ip: 192.168.1.25, mask: 255.255.255.0, gw: 192.168.1.1
I (906681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (909681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (912681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (915681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (918681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (921681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (923241) wifi:bcn_timout,ap_probe_send_start
I (924681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (925751) wifi:ap_probe_send over, resett wifi status to disassoc
I (925751) wifi:state: run -> init (c800)
I (925751) wifi:pm stop, total sleep time: 13884939 us / 25096644 us

I (925751) wifi:new:<4,0>, old:<4,0>, ap:<255,255>, sta:<4,0>, prof:1
I (925761) BLE_PROV: Disconnected. Connecting to the AP again...
E (925771) aws_iot: failed! mbedtls_ssl_handshake returned -0x2700
E (925771) aws_iot:     Unable to verify the server's certificate. 
I (925891) wifi:new:<4,0>, old:<4,0>, ap:<255,255>, sta:<4,0>, prof:1
I (925891) wifi:state: init -> auth (b0)
I (925891) wifi:state: auth -> assoc (0)
I (925901) wifi:state: assoc -> run (10)
I (925941) wifi:connected with TALKTALK4B89FC, aid = 1, channel 4, BW20, bssid = d0:c6:5b:4b:8a:0c
I (925941) wifi:security: WPA2-PSK, phy: bgn, rssi: -61
I (925951) wifi:pm start, type: 1

I (925951) wifi:AP's beacon interval = 102400 us, DTIM period = 1
I (927681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (928801) BLE_PROV: Connected with IP Address:192.168.1.25
I (928801) esp_netif_handlers: sta ip: 192.168.1.25, mask: 255.255.255.0, gw: 192.168.1.1
I (930681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (933681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (936681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (939681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (942681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (945681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (948681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (951681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (954681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (957681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (960681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (963681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (964511) wifi:bcn_timout,ap_probe_send_start
I (966681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (967011) wifi:ap_probe_send over, resett wifi status to disassoc
I (967011) wifi:state: run -> init (c800)
I (967011) wifi:pm stop, total sleep time: 26345448 us / 41062945 us

I (967021) wifi:new:<4,0>, old:<4,0>, ap:<255,255>, sta:<4,0>, prof:1
I (967021) BLE_PROV: Disconnected. Connecting to the AP again...
E (967031) aws_iot: failed! mbedtls_ssl_handshake returned -0x2700
E (967041) aws_iot:     Unable to verify the server's certificate. 
I (967151) wifi:new:<4,0>, old:<4,0>, ap:<255,255>, sta:<4,0>, prof:1
I (967151) wifi:state: init -> auth (b0)
I (967161) wifi:state: auth -> assoc (0)
I (967171) wifi:state: assoc -> run (10)
I (967201) wifi:connected with TALKTALK4B89FC, aid = 1, channel 4, BW20, bssid = d0:c6:5b:4b:8a:0c
I (967201) wifi:security: WPA2-PSK, phy: bgn, rssi: -61
I (967211) wifi:pm start, type: 1

I (967211) wifi:AP's beacon interval = 102400 us, DTIM period = 1
I (969681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (970301) BLE_PROV: Connected with IP Address:192.168.1.25
I (970301) esp_netif_handlers: sta ip: 192.168.1.25, mask: 255.255.255.0, gw: 192.168.1.1
I (972681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (975681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (978681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (981681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (984681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (987681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (990681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (993681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (996681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (999681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (1002681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (1005681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (1008681) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (1011691) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (1014691) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (1017691) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (1020691) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (1023691) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (1026691) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (1029691) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (1032691) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (1035691) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

E (1036121) aws_iot: failed! mbedtls_ssl_handshake returned -0x6800
I (1038691) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (1041691) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (1044691) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (1047691) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (1050691) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (1053691) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (1056691) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (1059691) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (1062691) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (1065691) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (1068691) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (1071691) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (1074691) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (1077691) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (1080691) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (1083691) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (1086691) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (1089691) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":25,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

I (1092691) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}
```

**Comment by chegewara** [comment]
I still dont see any issues here. Esp32 is disconnected from router which probably is caused by wifi sleep mode set up.
This is not an issue and place to get help is on forum.

**Comment by rhr407** [comment]
So it means that it's not an internal issue. This could pertain to the wifi router?

**Comment by chegewara** [comment]
Try to add this line to your code:
`esp_wifi_set_ps(WIFI_PS_NONE);`

Of course i may miss something, but to me it looks like normal behavior. esp32 is disconnected from AP, because you dont check in code if wifi is connected, you see mbedtls error, then after reconnecting to wifi AP you see shadow is sent again.

**Comment by AxelLin** [comment]
> I still dont see any issues here. Esp32 is disconnected from router which probably is caused by wifi sleep mode set up.

Do you mean if setting wifi sleep mode, esp32 may be disconnected from router?
In what kind of condition the esp32 will be disconnected due to wifi sleep mode?
I don't find document mentioned such behavior.

**Comment by AxelLin** [comment]
> ```c++
> run -> init (c800)
> ```
The disconnect reason is WIFI_REASON_BEACON_TIMEOUT, and the RSSI value seems between -61 ~ -62.
https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-guides/wifi.html#wi-fi-beacon-timeout

**Comment by rhr407** [comment]
Is this something a normal behavior by idf?

**Comment by ncohen17** [comment]
Any updates regarding this issue ? Is there a solution to fix it ?

**Comment by mahavirj** [comment]
@rhr407

-  Can you please confirm this issue with any of stable IDF release, e.g. IDF v4.0 or v4.1?
-  Also did you try with AP of different make/model, if possible would also request to log RSSI value using  `esp_wifi_sta_get_ap_info` API.

Might be relevant: https://github.com/espressif/esp-idf/issues/5915

**Comment by rhr407** [comment]
I have tested this on v4.1 which is a stable version and got the following error although it's not that frequent. Now the tls error has gone but this one is still persistent.

```
I (767733) RX_UART_TASK: Read 99 bytes: '{"DT":2,"T":40,"S":0,"P":10.1,"F":999.9,"E":999.9,"U":11.1,"A":999,"LO":-122.200682,"LA":37.692625}

E (769663) aws-shadow: Update timed out
W (769873) wifi:alloc eb len=24 type=3 fail, heap:32

W (769873) wifi:m f null

I (770663) wifi:bcn_timout,ap_probe_send_start
ESP_ERROR_CHECK failed: esp_err_t 0x101 (ESP_ERR_NO_MEM) at 0x4008df20
0x4008df20: _esp_error_check_failed at /home/riz/esp/esp-idf/components/esp32/panic.c:739

file: "/home/riz/esp/esp-idf/components/esp_common/src/ets_timer_legacy.c" line 70
func: ets_timer_setfn
expression: esp_timer_create(&create_args, (esp_timer_handle_t*)&(ptimer->timer_arg))

ELF file SHA256: a98f5941951a2c56

Backtrace: 0x4008d971:0x3ffd04f0 0x4008df23:0x3ffd0510 0x4010b085:0x3ffd0530 0x400d3de1:0x3ffd0560 0x40163980:0x3ffd0580 0x40160d58:0x3ffd05a0 0x40160ebe:0x3ffd05c0 0x40097692:0x3ffd05e0 0x4008f1d1:0x3ffd0610
0x4008d971: invoke_abort at /home/riz/esp/esp-idf/components/esp32/panic.c:157

0x4008df23: _esp_error_check_failed at /home/riz/esp/esp-idf/components/esp32/panic.c:740

0x4010b085: ets_timer_setfn at /home/riz/esp/esp-idf/components/esp_common/src/ets_time
```

**Comment by mahavirj** [comment]
@rhr407 

Good that earlier issue is no more seen with stable release!

> ESP_ERROR_CHECK failed: esp_err_t 0x101 (ESP_ERR_NO_MEM) at 0x4008df20

This points to problem where no memory is available for allocation or its too fragmented to satisfy current block size requirement. Can you please log heap statistics (free memory, largest free block, minimum free heap) parameters at this point? Post on memory analysis [here](https://medium.com/the-esp-journal/esp32-memory-analysis-case-study-eacc75fe5431) can be useful as reference.

Mahavir

**Comment by rhr407** [comment]
I am surprised that I could not find the data type 'heap_task_stat_t' anywhere in files of esp-idf folder and compiler giving me the following error.

```
esp-idf/aws/CMakeFiles/__idf_aws.dir/aws.c.obj -MF esp-idf/aws/CMakeFiles/__idf_aws.dir/aws.c.obj.d -o esp-idf/aws/CMakeFiles/__idf_aws.dir/aws.c.obj   -c ../components/aws/aws.c
../components/aws/aws.c: In function 'esp_dump_per_task_heap_info':
../components/aws/aws.c:33:5: error: unknown type name 'heap_task_stat_t'; did you mean 'esp_eth_state_t'?
     heap_task_stat_t tstat = {};
     ^~~~~~~~~~~~~~~~
     esp_eth_state_t
../components/aws/aws.c:33:30: error: empty scalar initializer
     heap_task_stat_t tstat = {};
                              ^
../components/aws/aws.c:33:30: note: (near initialization for 'tstat')
../components/aws/aws.c:38:22: error: implicit declaration of function 'heap_caps_get_next_task_stat'; did you mean 'heap_caps_get_total_size'? [-Werror=implicit-function-declaration]
         size_t ret = heap_caps_get_next_task_stat(&tstat, begin);
                      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~
                      heap_caps_get_total_size
../components/aws/aws.c:43:38: error: request for member 'task' in something not a structure or union
         const char *task_name = tstat.task ? pcTaskGetTaskName(tstat.task) : "Pre-Scheduler allocs";
                                      ^
../components/aws/aws.c:43:69: error: request for member 'task' in something not a structure or union
         const char *task_name = tstat.task ? pcTaskGetTaskName(tstat.task) : "Pre-Scheduler allocs";
                                                                     ^
../components/aws/aws.c:46:36: error: request for member 'peak' in something not a structure or union
                    task_name, tstat.peak[0], tstat.peak[1]);
                                    ^
../components/aws/aws.c:46:51: error: request for member 'peak' in something not a structure or union
                    task_name, tstat.peak[0], tstat.peak[1]);
                                                   ^
../components/aws/aws.c: In function 'aws_shadow':
../components/aws/aws.c:153:31: warning: variable 'connectParams' set but not used [-Wunused-but-set-variable]
     IoT_Client_Connect_Params connectParams = iotClientConnectParamsDefault;
                               ^~~~~~~~~~~~~
../components/aws/aws.c:152:28: warning: variable 'mqttInitParams' set but not used [-Wunused-but-set-variable]
     IoT_Client_Init_Params mqttInitParams = iotClientInitParamsDefault;
                            ^~~~~~~~~~~~~~
cc1: some warnings being treated as errors
ninja: build stopped: subcommand failed.
ninja failed with exit code 1
```

**Comment by mahavirj** [comment]
@rhr407 

There is note in article about IDF repository pointer. Alternatively you may use example https://github.com/espressif/esp-idf/tree/master/examples/system/heap_task_tracking. Also please use APIs `esp_get_free_heap_size`, `esp_get_minimum_free_heap_size` to log heap statistics.

**Comment by mahavirj** [comment]
@rhr407 Any update here? Can we close the issue?

**Comment by rhr407** [comment]
I am really sorry for not responding in a while. I will put the code on test and let you know in a couple of days.

**Comment by rhr407** [comment]
I would request to please close this issue.
